src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -451,9 +476,9 @@ private static SourceLocation GetAttributeNameStartLocation(SyntaxTreeNode node)
             if (builder != null)
             {
                 // If the attribute was requested by a tag helper but the corresponding property was not a string,
-                // then treat its value as code. A non-string value can be any C# value so we need to ensure the
-                // SyntaxTreeNode reflects that.
-                if (isBoundNonStringAttribute)
+                // then treat its value as code if it was previously markup. A non-string value can be any C# value so
+                // we need to ensure the SyntaxTreeNode reflects that.
+                if (isBoundNonStringAttribute && builder.Kind == SpanKind.Markup)
```

Comment:
Are Razor comments allowed inside an attribute?
Think meta code is not legal here. Right?
Why leave transitions as-is?